util/src/validate.ts

```diff
@@ -185,34 +197,55 @@ function checkVertices(ids: string[], protocolPath: string): void {
 function checkEdges(ids: string[], protocolPath: string): void {
 	let outputMessage: string | undefined;
 	const program: TJS.Program = TJS.getProgramFromFiles([protocolPath]);
-	const edgeSchema: TJS.Definition | null = TJS.generateSchema(program, 'Edge', { required: true, noExtraProps: true });
 	let count: number = 1;
 	const length: number = ids.length;
 
+	const e11Schema = TJS.generateSchema(program, 'E11', { required: true, noExtraProps: true });
+	const e1NSchema = TJS.generateSchema(program, 'E1N', { required: true, noExtraProps: true });
+	const itemSchema = TJS.generateSchema(program, 'ItemEdge', { required: true, noExtraProps: true });
+
 	ids.forEach((key: string) => {
 		const edge: LSIF.Edge = edges[key].element as LSIF.Edge;
 		outputMessage = `Verifying edge ${count} of ${length}...`;
 		process.stdout.write(`${outputMessage}\r`);
 		count++;
 
-		const validation: ValidatorResult = validateSchema(edge, edgeSchema);
-		if (!validation.valid) {
-			let errorMessage: string | undefined;
+		if ((!LSIF.Edge.is11(edge) && !(LSIF.Edge.is1N(edge))) || edge.outV === undefined) {
+			// This error was caught before
+			return;
+		}
+
+		if (edge.label === undefined || !Object.values(LSIF.EdgeLabels).includes(edge.label)) {
+			errors.push(new Error(edges[key].element, edge.label ? `requires property "label"` : `unknown label`));
 			edges[key].invalidate();
+			return;
+		}
 
-			if ((LSIF.Edge.is11(edge) && edge.inV === undefined) ||
-				LSIF.Edge.is1N(edge) && edge.inVs === undefined || edge.outV === undefined) {
-				// This error was caught before
-				return;
-			}
+		let validation: ValidatorResult;
+		switch (edge.label) {
+			case LSIF.EdgeLabels.item:
+				validation = validateSchema(edge, itemSchema);
+				break;
+			case LSIF.EdgeLabels.contains:
+				validation = validateSchema(edge, e1NSchema);
+				break;
+			default:
+				validation = validateSchema(edge, e11Schema);
+		}
 
-			if (edge.label === undefined) {
-				errorMessage = `requires property "label"`;
-			} else if (!Object.values(LSIF.EdgeLabels)
-						.includes(edge.label)) {
-				errorMessage = `unknown label`;
-			}
-			errors.push(new Error(edges[key].element, errorMessage!));
+		const validationErrors: ValidationError[] = validation.errors.filter((error) => error.property === 'instance');
+		if (validationErrors.length > 0) {
+			edges[key].invalidate();
+			let errorMessage: string = '';
+
+			validationErrors.forEach((error: ValidationError, index: number) => {
+				if (index > 0) {
+					errorMessage += '; ';
+				}
+				errorMessage += `${error.message}`;
+			});
```

Comment:
You can use `array.join("; ");` here instead on manually concatenating the string.

Reply:
Thank you for catching this! This is also true for another similar code snippet on the `checkVertices` function. I've updated both accordingly.